### DISCUSSION TITLE
[@types/mixpanel] does not allow string types for the `Mixpanel.Config.track_pageview` parameter

### DISCUSSION THREAD

**Discussion by cravindra** [DefinitelyTyped/DefinitelyTyped#69062] [Issues with a @types package]
I was trying to use one of the string values [documented in the mixpanel docs](https://docs.mixpanel.com/docs/tracking-methods/sdks/javascript#tracking-page-views) and ran into a typescript error:

```
Type 'string' is not assignable to type 'boolean | undefined'.ts(2322)
```

Code that caused it:

```
export default function initAnalytics() {
  mixpanel.init(MIXPANEL_TOKEN, {
    debug: false,
    track_pageview: 'url-with-path-and-query-string',
  });
}
```
What I did:

- I assumed that my types were out of date and updated to the latest version `2.14.8` but the issue persisted.
- I searched for any open issued with 'mixpanel' in them but found none
- I searched for any open PRs with 'mixpanel' in the but found none
- [Looked up the code](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0557b87e9ee2e1d1bb898cab290acb1d8dfc2563/types/mixpanel/index.d.ts#L589) on the `master` branch and verified that they type was as I was seeing it

Current code:

```
 /**
   * Should we track a page view on page load
   *
   * @default true
 */
track_pageview?: boolean | undefined;
```

Proposed new typing based on the docs:
```
 /**
   * Should we track a page view on page load
   *
   * @default undefined
 */
track_pageview?: "url-with-path" | "url-with-path-and-query-string" | "full-url" | boolean | undefined;
```

Happy to raise a PR if the above makes sense.

PS: The page tracking is turned off by default according to the docs so the default value also should be changed to undefined

[Excerpts from the docs](https://docs.mixpanel.com/docs/tracking-methods/sdks/javascript#tracking-page-views):

> Page view tracking is turned off by default. Page view events can be added automatically on every page load by using the track_pageview option to the mixpanel.init call, like so:

```
mixpanel.init('YOUR_TOKEN', {track_pageview: true});
```

> The default track_pageview setting does not auto-track page views in single-page applications. For tracking dynamic page views in single-page applications, the track_pageview option can also accept the following values for tracking.

```
// Track when the path changes, ignoring any query string or hash changes
mixpanel.init('YOUR_TOKEN', {track_pageview: "url-with-path"});
 
// Track when the path or query string change, ignoring hash changes
mixpanel.init('YOUR_TOKEN', {track_pageview: "url-with-path-and-query-string"});
 
// Track any URL changes in the path, query string, or hash
mixpanel.init('YOUR_TOKEN', {track_pageview: "full-url"});
```

[a comment by an automated account is not shown]

**Comment by ghost** [comment]
Fixed in #69157
